modules/fedimint-mint-server/src/lib.rs

```diff
@@ -363,15 +361,143 @@ impl ServerModule for Mint {
         consensus_items: Vec<(PeerId, MintConsensusItem)>,
         _consensus_peers: &BTreeSet<PeerId>,
     ) -> Vec<PeerId> {
-        for (peer, consensus_item) in consensus_items {
-            self.process_partial_signature(
-                dbtx,
-                peer,
-                consensus_item.out_point,
-                consensus_item.signatures,
+        for (peer_id, consensus_item) in consensus_items {
+            let out_point = consensus_item.out_point;
+            let signatures = consensus_item.signatures;
+
+            // check if we already obtained the blinded threshold signature for this
+            // out_point
+            if dbtx.get_value(&OutputOutcomeKey(out_point)).await.is_some() {
+                continue;
+            }
+
+            // check if we already have a valid signature share for this peer
+            if dbtx
+                .get_value(&ReceivedPartialSignatureKey(out_point, peer_id))
+                .await
+                .is_some()
+            {
+                continue;
+            }
+
+            // check if are currently collecting signature shares for this out_point
+            let our_contribution = match dbtx
+                .get_value(&ProposedPartialSignatureKey(out_point))
+                .await
+            {
+                Some(contribution) => contribution,
+                None => {
+                    warn!("Received signature share for non-existent out point");
```

Comment:
`warn` is correct because we already checked if there is a finalized signature already, otherwise this warning would occur every time all n guardians submit shares since the last `n-t` shares wouldn't be needed.